app/Movie.php

```diff
@@ -9,4 +9,63 @@ class Movie extends Model
     protected $fillable = [
         'title', 'description', 'image_url', 'genre_id', 'user_id'
     ];
+
+    /**
+     * The accessors to append to the model's array form.
+     *
+     * @var array
+     */
+    protected $appends = ['total_likes', 'liked_by_user', 'total_dislikes', 'disliked_by_user'];
+
+    function likes()
+    {
+        return $this->hasMany(Like::class);
+    }
+
+    function dislikes()
+    {
+        return $this->hasMany(Dislike::class);
+    }
+
+    /**
+     * Get all of the likes for the movie.
+     */
+    public function getTotalLikesAttribute()
+    {
+        return $this->likes()->count();
+    }
+
+    /**
+     * Checks if current user liked movie.
+     */
+    public function getLikedByUserAttribute()
+    {
+        if (auth()->user() != null) {
+            $user_id = auth()->user()->id;
```

Comment:
Ne znam, cudno mi je ovo, ne bih zavisio od auth()->user() ovde, kreira Service Locator Antipattern, kreira svuda generalno sa auth()->user() ali laravel, samo bih izbegao da to radimo unutar modela, radije bih da dobijes user kroz argument ali to ovde nije moguce :D
Mozes da napravis scope https://laravel.com/docs/7.x/eloquent#query-scopes ali ne globalni neko lokalni za njega, tipa da dobijes usera kao arugment onda